app/src/main/java/com/example/playlistmakerag/search/ui/view_models/SearchViewModel.kt

```diff
@@ -1,25 +1,33 @@
 package com.example.playlistmakerag.search.ui.view_models
 
+import android.content.Context
 import android.content.SharedPreferences
+import android.os.Handler
+import android.os.Looper
 import android.view.View
+import androidx.appcompat.app.AppCompatActivity
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModelProvider
 import androidx.lifecycle.viewmodel.initializer
 import androidx.lifecycle.viewmodel.viewModelFactory
 import com.example.playlistmakerag.app.App
+import com.example.playlistmakerag.app.PREFERENCES
 import com.example.playlistmakerag.player.domain.models.Track
 import com.example.playlistmakerag.player.data.dto.TrackResponse
 import com.example.playlistmakerag.search.data.SearchHistory
 import com.example.playlistmakerag.search.domain.SearchInteractor
 import com.example.playlistmakerag.search.domain.impl.SearchInteractorImpl
+import com.example.playlistmakerag.search.ui.SearchActivity
 import retrofit2.Response
 
 
 class SearchViewModel(private val interactor: SearchInteractorImpl) : ViewModel() {
```

Comment:
Взаимодействие между слоями должно происходить через интерфейсы:
```suggestion
class SearchViewModel(private val interactor: SearchInteractor) : ViewModel() {
``` 

В SettingsViewModel также укажи интерфейсы, а не имплементации